Microsoft.Toolkit.Uwp.UI.Controls/TokenizingTextBox/TokenizingTextBox.cs

```diff
@@ -214,7 +343,8 @@ private void RemoveToken(TokenizingTextBoxItem item)
                 return;
             }
 
-            SelectedItems.Remove(item.Content);
+            SelectedItems.Remove(item);
```

Comment:
Ah, missed this, these should be `item.Content` the Selected and Tokenized lists should be of the data type of the object, not the TokenizingTextBoxItem container, you can see this is how base collections work like ListView, they return the raw items.